### PR TITLE
Support EXISTS(subquery)

### LINKED ISSUE #6852: WHERE EXISTS ( ... subquery ...) not supported and error message is confusing
https://www.w3schools.com/sql/sql_exists.asp
ClickHouse doesn't support such a queries and thinks that 'exists' is a column name:
```
select * from system.one where exists (select 1);

Expected one of: LIKE, GLOBAL NOT IN, AS, IS, OR, QuestionMark, BETWEEN, NOT LIKE, AND, Comma, alias, IN, Dot, NOT, Arrow, token, NOT IN, GLOBAL IN

select * from system.one where exists;

Received exception from server (version 19.14.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'exists' while processing query: 'SELECT dummy FROM system.one WHERE exists', required columns: 'dummy' 'exists', source columns: 'dummy'. 

0 rows in set. Elapsed: 0.011 sec. 
```

PS. Not important, issue is just to document that behaviour.

### PR DESCRIPTION
**Pull request by kssenii**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support EXISTS(subquery). Closes https://github.com/ClickHouse/ClickHouse/issues/6852.

### CODE REVIEW

File: src/Interpreters/TreeRewriter.cpp [hunk 1]
```diff
@@ -250,6 +250,61 @@ using CustomizeAggregateFunctionsOrNullVisitor = InDepthNodeVisitor<OneTypeMatch
...
+    void visit(ASTFunction & func, ASTPtr) const
+    {
+        bool exists_expression = func.name == "exists"
+            && func.arguments && func.arguments->children.size() == 1
+            && typeid_cast<const ASTSubquery *>(func.arguments->children[0].get());
+
+        if (!exists_expression)
+            return;
+
+        /// EXISTS(subquery) --> 1 IN (SELECT 1 FROM subquery LIMIT 1)
```
Comment (alexey-milovidov) [inline src/Interpreters/TreeRewriter.cpp:267]: Does LIMIT pushdown work alright?
CC @KochetovNicolai

Comment (alexey-milovidov) [reply src/Interpreters/TreeRewriter.cpp:267]: LGTM nevertheless.

**Review by alexey-milovidov — APPROVED** [review]

File: src/Interpreters/TreeRewriter.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Interpreters/TreeRewriter.cpp:267]: It should.
Some things may be still improved for exists, e.g. we don't have to wait for `GROUP BY`

**Comment by alexey-milovidov** [comment]
Cannot merge, tests have failed :(

**Comment by kssenii** [comment]
`Integration tests (thread) — fail: 67` -- docker problems..
Other failures also not related.

**Comment by ka1bi4** [comment]
Internal documentation ticket: DOCSUP-17653

**Pull request merged**